File: LeetCode/cpp/1038.cpp
class Solution {
public:
    TreeNode *bstToGst(TreeNode *root) {
        Traverse(root, 0);
        return root;
    }

    int Traverse(TreeNode *node, int val) {
        if (!node)
            return val;
        int right = Traverse(node->right, val);
        node->val += right;
        int left = Traverse(node->left, node->val);
        return left;
    }
};

File: LeetCode/python3/1038.py
class Solution:
    def bstToGst(self, root: TreeNode) -> TreeNode:
        self.Traverse(root, 0)
        return root
    
    def Traverse(self, node: TreeNode, val: int) -> int:
        if not node:
            return val
        right = self.Traverse(node.right, val)
        node.val += right
        left = self.Traverse(node.left, node.val)
        return left